rcl_interfaces/msg/FloatingPointRange.msg

```diff
@@ -0,0 +1,14 @@
+# Represents bounds and a step value for a floating point typed parameter.
+
+# Lower bound for valid values, inclusive.
+double minimum_value
+
+# Upper bound for valid values, inclusive.
+double maximum_value
```

Comment:
Should we allow minimum value to be greater than and/or equal to maximum value?

Reply:
I would advocate for not supporting inverting min and max values. If I told you a min values of 10 and a max value of 9 you could not pick a value that satisfies those constraints. I think that if you want to control the UI that closely you should use another mechanism to lay out the visualization for your customized application visualization.

Reply:
I agree that having invertible sliders (e.g. 2-1-0 rather than 0-1-2) is desirable.

Then maybe we should consider different terminology, because minimum is less than or equal to maximum by definition in most situations. Maybe `from` and `to`?

Reply:
Ok, if we want an invertable slider, we should have a `bool invert_slider` in the description not in the constraint. However I don't believe that we should tangle that up in the constraint section. It's not related to the constraints. Constraint checking will have to be universally implemented and extra complexity there is not where we want things.

Reply:
I can buy the argument that we don't want extra complexity in the constraint enforcement, but on the other hand, you must check for this case anyways (we cannot prevent someone from passing a parameter descriptor which has a min > max), so you can choose to error in that case, or simply flip them for the purposes of validation.

I don't think we should embed something like `invert_slider` into the parameter descriptor. I see this discussion of using `from`/`to` instead as a way of changing what we're saying the constraint is in order to be more flexible, e.g. "the value is between the values from and to" versus "the value is greater than minimum and less than maximum".

This wouldn't work, however, if you wanted to bound just one side of the value, e.g. "greater than 5".  I suppose you could do what is done now, which is to select an arbitrarily large value. The other problem with bounding only one side is that rendering that as a slider is hard, because you don't know what scale to put the slider on.

My thinking is that these are actually two different kinds of constraints (mutually exclusive constraints but still different) which would be represented differently in a GUI. A range would be between two values, have a step size, and be represented by a slider. A "numeric constraint" would be minimum value or a maximum value and in the GUI that would simply be a text box that checked that the value entered didn't violate the constraint, i.e. no slider.

Reply:
MHO:

> I can buy the argument that we don't want extra complexity in the constraint enforcement, but on the other hand, you must check for this case anyways (we cannot prevent someone from passing a parameter descriptor which has a min > max), so you can choose to error in that case, or simply flip them for the purposes of validation.

 :+1: to using `from` / `to` naming and to this rationale. The only warning I see is that this approach seems a bit more error prone than using `min` / `max` and not allowing `max` < `min`, but if using invertible sliders is desirable then it's worth it to gain flexibility.

> This wouldn't work, however, if you wanted to bound just one side of the value, e.g. "greater than 5". I suppose you could do what is done now, which is to select an arbitrarily large value. The other problem with bounding only one side is that rendering that as a slider is hard, because you don't know what scale to put the slider on.

I think using arbitrarily large numbers should be enough for most use-cases. In the end you are still constrained by the max (or min) value the datatype can hold. For floats you should be able to use `inf` right?

Reply:
> I think using arbitrarily large numbers should be enough for most use-cases. In the end you are still constrained by the max (or min) value the datatype can hold. For floats you should be able to use inf right?
I agree

Reply:
In the case of "greater than 5", I still don't think you want a slider, unless you pick some arbitrarily large, but not too large, upper bound like 100. If you use `inf` or `INT64_MAX`, then you probably end up with a useless slider. That's why I think they're two different kinds of constraints.